src/MTGCore/Services/SearchFilterMapper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using MTGCore.Services.Exceptions;
+using MTGCore.ViewModels;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace MTGCore.Services
+{
+    public class SearchFilterMapper : ISearchFilterMapper
+    {
+        private Dictionary<string, string> filterParameters = new Dictionary<string, string>();
+
+        private Dictionary<Type, Delegate> typeProcessorMap = new Dictionary<Type, Delegate>() 
+        {
+            { typeof(List<string>), new Func<PropertyInfo,SearchFilter,string>((x,y) => {return StringListResolver.ValueToString(x,y);}) },
+            { typeof(string), new Func<PropertyInfo,SearchFilter,string>((x,y) => {return StringResolver.ValueToString(x,y);}) }
+        };
+
+        public Dictionary<string, string> map(SearchFilter filter)
+        {
+
+            PropertyInfo[] properties = typeof(SearchFilterWithColours).GetProperties();
+            foreach (PropertyInfo property in properties)
+            {
+                Type type = property.PropertyType;
+
+                var result = typeProcessorMap[type].DynamicInvoke(property, filter);
```

Comment:
Currently, if the type doesn't exist in the VM, an exception would be thrown as the key would not be found in the dictionary. I'm not sure if we'd want to handle that exception, just in case someone adds a property in the future.